caddyhttp/httpserver/plugin.go

```diff
@@ -368,19 +377,24 @@ func groupSiteConfigsByListenAddr(configs []*SiteConfig) (map[string][]*SiteConf
 
 	for _, conf := range configs {
 		// We would add a special case here so that localhost addresses
-		// bind to 127.0.0.1 if conf.ListenHost is not already set, which
+		// bind to 127.0.0.1 if conf.ListenHosts is not already set, which
 		// would prevent outsiders from even connecting; but that was problematic:
 		// https://caddy.community/t/wildcard-virtual-domains-with-wildcard-roots/221/5?u=matt
 
 		if conf.Addr.Port == "" {
 			conf.Addr.Port = Port
 		}
-		addr, err := net.ResolveTCPAddr("tcp", net.JoinHostPort(conf.ListenHost, conf.Addr.Port))
-		if err != nil {
-			return nil, err
+		if len(conf.ListenHosts) == 0 {
+			conf.ListenHosts = append(conf.ListenHosts, "127.0.0.1")
```

Comment:
What is the reason for this?

Reply:
oh, I misread the comment above. It said 'We would add a special case here so that localhost addresses bind to 127.0.0.1 if conf.ListenHost is not already set', but it also said 'that was problematic'. I will change it.